v3-sdk/swap-and-add-liquidity/src/example/Example.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import './Example.css'
+import { Percent, CurrencyAmount, Fraction } from '@uniswap/sdk-core'
+import { Environment, CurrentConfig } from '../config'
+import { getCurrencyBalance } from '../libs/wallet'
+import {
+  getPositionIds,
+  constructPosition,
+  mintPosition,
+} from '../libs/liquidity'
+import {
+  connectBrowserExtensionWallet,
+  getProvider,
+  TransactionState,
+  sendTransaction,
+  getWalletAddress,
+  getMainnetProvider,
+} from '../libs/providers'
+import {
+  NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+  V3_SWAP_ROUTER_ADDRESS,
+} from '../libs/constants'
+import { fromReadableAmount } from '../libs/conversion'
+import {
+  AlphaRouter,
+  SwapAndAddConfig,
+  SwapAndAddOptions,
+  SwapType,
+  SwapToRatioStatus,
+} from '@uniswap/smart-order-router'
+
+async function swapAndAddLiquidity(
+  positionId: number
+): Promise<TransactionState> {
+  const address = getWalletAddress()
+  const provider = getProvider()
+  if (!address || !provider) {
+    return TransactionState.Failed
+  }
+
+  const router = new AlphaRouter({ chainId: 1, provider: getMainnetProvider() })
+
+  const swapAndAddConfig: SwapAndAddConfig = {
+    ratioErrorTolerance: new Fraction(1, 100),
+    maxIterations: 6,
+  }
+
+  const swapAndAddOptions: SwapAndAddOptions = {
+    swapOptions: {
+      type: SwapType.SWAP_ROUTER_02,
+      recipient: address,
+      slippageTolerance: new Percent(5, 100),
+      deadline: 60 * 20,
+    },
+    addLiquidityOptions: {
+      tokenId: positionId,
+    },
+  }
+
+  const currentPosition = await constructPosition(
+    CurrencyAmount.fromRawAmount(
+      CurrentConfig.tokens.token0,
+      fromReadableAmount(
+        CurrentConfig.tokens.token0Amount,
+        CurrentConfig.tokens.token0.decimals
+      )
+    ),
+    CurrencyAmount.fromRawAmount(
+      CurrentConfig.tokens.token1,
+      fromReadableAmount(
+        CurrentConfig.tokens.token1Amount,
+        CurrentConfig.tokens.token1.decimals
+      )
+    )
+  )
+
+  const routeToRatioResponse = await router.routeToRatio(
+    CurrencyAmount.fromRawAmount(
+      CurrentConfig.tokens.token1,
+      fromReadableAmount(
+        CurrentConfig.tokens.token1Amount,
+        CurrentConfig.tokens.token1.decimals
+      )
+    ),
+    CurrencyAmount.fromRawAmount(
+      CurrentConfig.tokens.token0,
+      fromReadableAmount(
+        CurrentConfig.tokens.token0Amount,
+        CurrentConfig.tokens.token0.decimals
+      )
+    ),
```

Comment:
We do these twice, extract as setup constants? Will also make more readable for the guide!